pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTRecordDeclaration.java

```diff
@@ -55,12 +55,6 @@ public boolean isFindBoundary() {
         return isNested() || isLocal();
     }
 
-    @Override
-    public boolean isFinal() {
```

Comment:
I think we should keep that here and add a new method `isSyntacticallyFinal` that returns `super.isFinal` (and can be used in UnnecessaryModifier). Otherwise the contract of `isFinal` is not respected

Reply:
done